juriscraper/opinions/united_states/state/conn.py

```diff
@@ -23,30 +23,69 @@ class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.court_id = self.__module__
-        self.year = date.today().strftime("%y")
-        self.url = f"http://www.jud.ct.gov/external/supapp/archiveAROsup{self.year}.htm"
+        year = date.today().strftime("%y")
+        self.url = (
+            f"http://www.jud.ct.gov/external/supapp/archiveAROsup{year}.htm"
+        )
         self.status = "Published"
+        self.cipher = "AES256-SHA256"
+        self.set_custom_adapter(self.cipher)
+
+    @staticmethod
+    def find_docket_numbers(text: str) -> str:
+        """Find docket numbers in row
+
+        :param text: the row text
+        :return: return docket numbers
+        """
+        matches = re.findall(r"SC\d+", text)
+        return ", ".join(matches)
+
+    @staticmethod
+    def remove_excess_whitespace(text: str) -> str:
+        return re.sub(r"\s+", " ", text)
+
+    @staticmethod
+    def find_published_date(row):
+        closest_published = row.xpath(
+            "preceding::*[contains(., 'Published')][1]"
+        )
+        date_str = closest_published[0].text_content()
+        m = re.search(
+            r"(\b\d{1,2}/\d{1,2}/\d{4}\b)|(\b(?:January|February|March|April|May|June|July|August|September|October|November|December) \d{1,2}, \d{4}\b)",
+            date_str,
+        )
+        return m.groups()[0] if m.groups()[0] else m.groups()[1]
 
     def _process_html(self) -> None:
         """Process the html and extract out the opinions
 
         :return: None
         """
-        for row in self.html.xpath("//ul/preceding-sibling::p"):
-            if "Published in the Law Journal" not in row.text_content():
-                continue
-            date = row.text_content().split(" ")[-1][:-1]
-            for document in row.xpath("following-sibling::ul[1]/li"):
-                output_string = re.sub(
-                    r"\s{2,}", " ", document.text_content()
-                ).strip()
-                name = output_string.split("-")[-1].strip()
-                docket = document.xpath(".//a")[0].text_content().split()[0]
-                self.cases.append(
-                    {
-                        "date": date,
-                        "url": document.xpath(".//a")[0].get("href"),
-                        "docket": docket,
-                        "name": name,
-                    }
+        pdf_xpath = "//table[@id='AutoNumber1']//a[contains(@href, '.pdf')]"
+        for row in self.html.xpath(pdf_xpath):
+            url = row.get("href")
+            link_text = row.text_content().strip()
+            full_row_text = row.getparent().text_content().strip()
+            if link_text == full_row_text:
+                # Fix for atypical bug found in an older year
+                full_row_text = (
+                    row.getparent().getparent().text_content().strip()
                 )
```

Comment:
Scraping 2020 is raising an InsanityException, since there is an empty case name

The fix works if you do the remove_excess_whitespace before the comparison (which will be done anyway) and do a str.strip('-') call. The failing values:
```
row.getparent().text_content().strip()
> 'SC20208, SC20209 \r\n\t\t\t\t\t-'
link_text
> 'SC20208, SC20209'
```

Reply:
In 2023 it is picking up a wrong case name (mixing the name of all the cases in that date), because all the info is inside \<a> tag
```
 {'date': '6/20/2023',
  'docket': 'SC20694',
  'name': 'SC20737 - Cohen v. Rossi SC20737 First Concurrence - Cohen v. Rossi '
          'SC20737 Second Concurrence - Cohen v. RossiSC20630 - Clark v. '
          'Waterford, Cohanzie Fire Dept.SC20630 Dissent - Clark v. Waterford, '
          'Cohanzie Fire Dept.',
  'url': 'http://www.jud.ct.gov/external/supapp/Cases/AROcr/CR346/346CR35.pdf'},
  ```
Both of these are from past years that probably are not of interest, but the missformating may appear again

Reply:
I reworked the code and I think ... I think ive resolved this issue 

```
52 Cohen v. Rossi (Second Concurrence)   http://www.jud.ct.gov/external/supapp/Cases/AROcr/CR346/346CR31B.pdf
53 Cohen v. Rossi (First Concurrence)    http://www.jud.ct.gov/external/supapp/Cases/AROcr/CR346/346CR31A.pdf
54 Cohen v. Rossi        http://www.jud.ct.gov/external/supapp/Cases/AROcr/CR346/346CR31.pdf
55 Clark v. Waterford, Cohanzie Fire Dept. (Dissent)     http://www.jud.ct.gov/external/supapp/Cases/AROcr/CR346/346CR33E.pdf
56 Clark v. Waterford, Cohanzie Fire Dept.       http://www.jud.ct.gov/external/supapp/Cases/AROcr/CR346/346CR33.pdf
```